### PR TITLE
Bug 1703874 - Nudge Existing Users to Set as Default Using Infobar

### PR DESCRIPTION
**Pull request by piatra**

### CODE REVIEW

**Comment by piatra** [comment]
Schema check for CFR buttons is "disabled" by this PR. Seems like the top schema definitions are not being read.
I've also tried 
```json
anyOf: [
  { the current definitions for buttons }
  { "type": "array" }
]
```
But it didn't seem to like mixing those two schemas together.
Thinking of making maybe a separate folder `cfr_infobar.yaml` with the modified schema. It's different enough that it's annoying to match both types of messages with 1 schema.

File: cfr.yaml [hunk 1]
```diff
@@ -104,3 +104,29 @@
...
+  trigger:
+    id: defaultBrowserCheck
+  template: infobar
+  frequency:
+    lifetime: 100
+    custom:
+      - period: 3024000000  # 86400000 * 35 (35 days to ms)
+        cap: 1
+  # Not default browser and user dismissed the default browser prompt
+  targeting: "!isDefaultBrowser && !'browser.shell.checkDefaultBrowser'|preferenceValue"
```
Comment (ncloudioj) [inline cfr.yaml:132]: I think Alex mentioned this "Avoid collision with existing user onboarding (MR1)" in the bug. Do we have a way to detect that?

**Review by ncloudioj — CHANGES_REQUESTED** [review]

File: cfr.yaml (hunk 1 above)
Comment (piatra) [reply cfr.yaml:132]: Coming up https://phabricator.services.mozilla.com/D112090

File: cfr.yaml [hunk 2]
```diff
@@ -104,3 +104,29 @@
...
+  trigger:
+    id: defaultBrowserCheck
+  template: infobar
+  frequency:
+    lifetime: 100
+    custom:
+      - period: 3024000000  # 86400000 * 35 (35 days to ms)
+        cap: 1
+  # Not default browser and user dismissed the default browser prompt
+  targeting: "!isDefaultBrowser && !'browser.shell.checkDefaultBrowser'|preferenceValue && isMajorUpgrade != true"
```
Comment (ncloudioj) [inline cfr.yaml:132]: nit: let's do `!isMajorUpgrade`

**Review by ncloudioj — APPROVED** [review]
LGTM with another nit.

Thanks!

File: cfr.yaml (hunk 2 above)
Comment (piatra) [reply cfr.yaml:132]: Doing explicit `!= true` to also cover the scenario of the attribute not being defined.

**Pull request merged**